### PR TITLE
virtio_fs: add hugepage memory backend

### PR DESCRIPTION
**Pull request by xiagao**
Add memory-backend-file with hugepage and numa.
ID: 2016895
Signed-off-by: Xiaoling Gao <xiagao@redhat.com>

### CODE REVIEW

**Comment by xiagao** [comment]
(1/3) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.auto.q35: PASS (185.93 s)
 (2/3) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.with_numa.basic_test.with_cache.auto.q35: PASS (179.75 s)
 (3/3) Host_RHEL.m8.u6.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.6.0.x86_64.io-github-autotest-qemu.virtio_fs_share_data.with_cache.always.q35: PASS (147.48 s)
RESULTS    : PASS 3 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0

**Comment by xiagao** [comment]
@YongxueHong @zhenyzha  Call for your review.thx

File: qemu/tests/cfg/virtio_fs_share_data_hugepage.cfg [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+            Windows:
+                cmd_dd = 'dd if=/dev/random of=%s bs=1M count=200'
+                cmd_md5 = "%s: && md5sum.exe %s"
+                cmd_new_folder = "%s && md ${test_folder}"
+                cmd_copy_file = "copy ${test_file} ${test_folder}"
+                cmd_rename_folder = "rename ${test_folder} ${test_folder}_tmp"
+                cmd_del_folder = "rd /q /s ${test_folder}_tmp"
+                cmd_check_folder = "dir ${test_folder}_tmp"
+    variants:
+        - @default:
```
Comment (zhenyzha) [inline qemu/tests/cfg/virtio_fs_share_data_hugepage.cfg:75]: The default mode is already included in other cases, do we still need this option?

**Comment by zhenyzha** [comment]
Other parts ok for me:
```
 (1/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.auto.arm64-pci: PASS (228.47 s)
 (2/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.always.arm64-pci: PASS (180.17 s)
 (3/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.basic_test.with_cache.none.arm64-pci: PASS (200.65 s)
 (4/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.with_numa.basic_test.with_cache.auto.arm64-pci: PASS (201.08 s)
 (5/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.with_numa.basic_test.with_cache.always.arm64-pci: PASS (184.83 s)
 (6/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.0.0.aarch64.io-github-autotest-qemu.virtio_fs_share_data_hugepage.with_numa.basic_test.with_cache.none.arm64-pci: PASS (194.38 s)
```

File: qemu/tests/cfg/virtio_fs_share_data_hugepage.cfg (hunk 1 above)
Comment (xiagao) [reply qemu/tests/cfg/virtio_fs_share_data_hugepage.cfg:75]: There is only one default in thie cfg file,I think it has no relation with other cfg files.

File: qemu/tests/virtio_fs_share_data.py [hunk 2]
```diff
@@ -126,8 +126,7 @@ def get_stdev(file):
     # st_dev check config
     cmd_get_stdev = params.get("cmd_get_stdev")
     nfs_mount_dst_name = params.get("nfs_mount_dst_name")
-
-    if cmd_xfstest:
+    if cmd_xfstest and not params.get("setup_hugepages"):
```
Comment (YongxueHong) [inline qemu/tests/virtio_fs_share_data.py:129]: this `not params.get("setup_hugepages"):` might not right.
if the `setup_hugepages = yes` and `setup_hugepages = no` is defined in cfg, this if block will awalys run.

**Review by YongxueHong — APPROVED** [review]

**Pull request merged**